### DISCUSSION TITLE
how to run both opacity and visible false together

### DISCUSSION THREAD

**Discussion by bobwatcherx** [flet-dev/flet#895] [Q&A]
I want when the description does an opacity animation and does visible False at the same time.  and the button below should shift up.  because the description part is missing.  is there a solution
![rere](https://user-images.githubusercontent.com/52072811/212861425-ceceea3f-639f-4374-a345-55b9722afcd6.png)

mycode:
```
from flet import *

def main(page:Page):

	title = Text("this is title ",size=30,weight="bold")
	desc = Text("Makes a control partially transparent. 0.0 - control is completely transparent, not painted at all. 1.0 (default) - a control is fully painted without any transparency.",
		width=300,
		animate_opacity=800

		)

	def showme(e):
		desc.opacity = 0
		page.update()
	page.add(
		title,
		desc,
		ElevatedButton("showme",
			on_click=showme

			)

		)
flet.app(target=main)
```

**Comment by bobwatcherx** [comment]
I want the animation to run smoothly and do visible false. because I have tried it however .  only visible is running.  but don't do a subtle opacity just yet

**Reply by ghost** [reply]
update the control after .opacity = 0, add time.sleep delay  with same lenght as opacity change, then change .visible to False.
Smooth fade out of  (visible) existence :)

```
    def showme(e):
        desc.opacity = 0
        desc.update()
        time.sleep(desc.animate_opacity/1000)
        desc.visible = False
        page.update()

    page.add(
        title,
        desc,
        ElevatedButton("showme",
                       on_click=showme

                       )
```
